nasa/earthdata-search
# EDSC-3880: Move colormaps state to Redux

# Overview

### What is the feature?

Added infrastructure to pull in colormaps into the redux store so that when we load in featuredCollections, we can more efficiently grab the colormap data instead of making a fetch request each time.

### What is the Solution?

- Added "metadata.colormaps" to the redux store to store colormap data.
- Added logic in featuredCollection that grabs all gibs tags and pulls in the colormaps if they are not in the metadata.colormaps object already
- Added 3 new actions to the newly created colorMap reducer 
  - SET_COLOR_MAPS_LOADING
  - SET_COLOR_MAPS_LOADED
  - ERRORED_COLOR_MAPS
- Added an 4 new actions to the newly created colorMap actions
  - setColorMapsLoading
  - setColorMapsLoaded
  - setColorMapsErrored
  - getColorMap
- Created a new ColorMapRequest object that connects to the colormaps api in the serverless api.
  - has the function getColorMap

### What areas of the application does this impact?

Impacts the focusedCollection part.

# Testing

### Reproduction steps

- **Environment for testing: Local/SIT data**
- **Collection to test with: PODAAC-GHGMR-4FJ04**

1. navigate to localhost:8080/search?ee=sit
2. paste in PODAAC-GHGMR-4FJ04 into the search bar and click enter.
3. Click on the PODAAC-GHGMR-4FJ04 collection.
4. Observe everything loads in correctly and that the legend on the top right loads in with the correct colors.
5. You can check using the developer tools that the state changes correctly when clicking on the collection.

### Attachments

<img width="1727" alt="image" src="https://github.com/nasa/earthdata-search/assets/8591291/ef441e62-2f73-4768-bc7e-cf750eb17c76">

# Checklist
